src/protocol_pro_2.cpp

```diff
@@ -156,6 +203,7 @@ void Pro2ProtocolObject::send_command(int sleeptime) {
     std::this_thread::sleep_for(std::chrono::milliseconds(sleeptime));
   }
 }
+
 int Pro2ProtocolObject::cycle_robot_mode() {
   if (robotmode_num_ < 2) {
```

Comment:
just an FYI a few things about this code could be improved. 

- Robot modes should be an enumeration in a header file, with explicit enumeration of each mode to an integer.
- Instead of using if/else to cycle the robot mode between [0, 1, 2], use a modulo '%' operator and enumerate the modes starting with 0. For example, robotmode = robotmode++ % number_of_modes or something similar. The modulo operator will achieve a counting sequence like 0, 1, 2, 0, 1, 2.
- 5 here is a magic number. It should be set in a header file here as LINEAR_JERK_LIMIT since it represents the limit to the derivative of acceleration.